libs/ui/src/lib/snackbar/snackbar.tsx

```diff
@@ -0,0 +1,33 @@
+import {Alert, AlertColor, Snackbar, SnackbarProps} from '@mui/material'
+import {createContext, useState} from 'react'
+
+export const SnackbarContext = createContext<{
+  setSnack: React.Dispatch<React.SetStateAction<Snack | undefined>>
+}>({
+  setSnack: () => {
+    return
+  }
+})
+
+export interface Snack {
+  message: string
+  severity?: AlertColor
+}
+
+export function SnackbarWithContextProvider({children, ...props}: SnackbarProps) {
```

Comment:
Currently this only supports 1 snackbar, maybe we want to do a push on it so we can have multiple. But not important for now of course. Works great!

Reply:
yeah. I wanted to implement that. But I gave up on it because of a really quirky behavior of the snackbar component. It is ok for now.